runtime/doc/nvim_terminal_emulator.txt

```diff
@@ -134,6 +134,11 @@ Example: >
   programs can set this by emitting an escape sequence.
 - |'channel'|  Terminal PTY |job-id|.  Can be used with |chansend()| to send
   input to the terminal.
+- *b:term_error*  Return code of the job that finished running in the terminal.
+  Analagous to |v:shell_error|.  Use this to automatically close terminal
+  buffers that finished without an error: >
+
+    autocmd TermClose * if !get(b:, 'term_error') | bdelete | endif
```

Comment:
Note: this particular example (using `:bdelete` without a `!`) depends on #15402 being merged.